@here/harp-mapview/lib/geometry/TileGeometryCreator.ts

```diff
@@ -1155,14 +1155,27 @@ export class TileGeometryCreator {
                             }
 
                             if (outlineTechnique.secondaryWidth !== undefined) {
-                                lineMaterial.lineWidth =
-                                    getPropertyValue(
-                                        outlineTechnique.secondaryWidth!,
-                                        mapView.zoomLevel,
-                                        mapView.pixelToWorld
-                                    ) *
-                                    unitFactor *
-                                    0.5;
+                                const techniqueLineWidth = getPropertyValue(
+                                    outlineTechnique.lineWidth!,
+                                    mapView.zoomLevel,
+                                    mapView.pixelToWorld
+                                );
+                                const techniqueSecondaryWidth = getPropertyValue(
+                                    outlineTechnique.secondaryWidth!,
+                                    mapView.zoomLevel,
+                                    mapView.pixelToWorld
+                                );
+                                const techniqueOpacity = getPropertyValue(
+                                    outlineTechnique.opacity,
+                                    mapView.zoomLevel
+                                );
+                                // hide outline when it's equal or smaller then line to avoid subpixel contour
+                                const lineWidth =
+                                    techniqueSecondaryWidth <= techniqueLineWidth &&
+                                    (techniqueOpacity === undefined || techniqueOpacity === 1)
+                                        ? 0
+                                        : techniqueSecondaryWidth;
+                                lineMaterial.lineWidth = lineWidth * unitFactor * 0.5;
```

Comment:
This makes sure that the line is not visible but will it also make sure that the line is not rendered?

Reply:
Currently we can't avoid rendering outlines, because we are setting lineWidth and color at onBeforeRender:
https://github.com/heremaps/harp.gl/blob/HARP-7109/%40here/harp-mapview/lib/geometry/TileGeometryCreator.ts#L1140

And if we switch visibility, it will stop rendering at all, even if lineWidth turns to positive value.